File: app/Global.java
import java.lang.reflect.Method;
import java.util.concurrent.TimeUnit;

import actors.AutoScaleModifier;
import actors.AutoScaleModifierImpl2;
import actors.PriceMonitor;
import actors.PriceMonitorImpl;
import akka.actor.TypedActor;
import akka.actor.TypedProps;
import play.Application;
import play.GlobalSettings;
import play.libs.Akka;
import scala.concurrent.duration.Duration;

public class Global extends GlobalSettings
{
    public void onStart(Application application)
    {
        AutoScaleModifier autoScaleModifier = TypedActor.get(Akka.system()).typedActorOf(new TypedProps<AutoScaleModifierImpl2>(AutoScaleModifier.class, AutoScaleModifierImpl2.class), "autoScaleModifier");
        PriceMonitor priceMonitor = TypedActor.get(Akka.system()).typedActorOf(new TypedProps<PriceMonitorImpl>(PriceMonitor.class, PriceMonitorImpl.class), "priceMonitor");
        Akka.system().scheduler().schedule(
                Duration.create(20, TimeUnit.SECONDS),
                Duration.create(60, TimeUnit.SECONDS),
                makeRunnable(autoScaleModifier, "monitorAutoScaleGroups"),
                Akka.system().dispatcher()
        );
        Akka.system().scheduler().schedule(
                Duration.create(0, TimeUnit.SECONDS),
                Duration.create(10, TimeUnit.SECONDS),
                makeRunnable(priceMonitor, "monitorSpotPrices"),
                Akka.system().dispatcher()
        );
    }
    
    private Runnable makeRunnable(final Object instance, final String methodName)
    {
        return new Runnable()
        {
            @Override
            public void run()
            {
                try
                {
                    Method method = instance.getClass().getMethod(methodName);
                    method.invoke(instance);
                }
                catch (Exception e)
                {
                    throw new RuntimeException();
                }
            }
        };
    }
}

File: app/Global.scala
import play.api.GlobalSettings
import play.api.Application
import actors.AutoScaleModifier
import play.libs.Akka
import actors.AutoScalingDataMonitor
import actors.PriceMonitor
import scala.concurrent.duration._
import scala.concurrent.ExecutionContext
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Future
import play.Logger
import util.NameHelper

object Global extends GlobalSettings {

    override def onStart(app: Application) {
        val priceMonitor = new PriceMonitor()
        val autoScalingDataMonitor = new AutoScalingDataMonitor()
        val autoScaleModifier = new AutoScaleModifier(autoScalingDataMonitor, priceMonitor)
        Akka.system.scheduler.schedule(0.seconds, 10.seconds) {
            wrapExceptionHandling(autoScaleModifier.monitorAutoScaleGroups)
        }
        Akka.system.scheduler.schedule(0.seconds, 60.seconds) {
            wrapExceptionHandling(priceMonitor.monitorSpotPrices)
        }
        Akka.system.scheduler.schedule(0.seconds, 15.seconds) {
            wrapExceptionHandling(priceMonitor.printPrices)
        }
        Akka.system.scheduler.schedule(0.seconds, 60.seconds) {
            wrapExceptionHandling(autoScalingDataMonitor.monitorAutoScalingData)
        }
    }

    private[this] def wrapExceptionHandling(action: => Unit) = {
        try {
            action
        }
        catch {
            case t:Throwable => {
                Logger.error("Error running scheduled task", t)
                throw t
            }
        }
    }
}